lib/elixir/lib/calendar/iso.ex

```diff
@@ -216,6 +216,22 @@ defmodule Calendar.ISO do
       ]
     end
 
+  [match_big_year, guard_big_year, match_big_year_rest, guard_big_year_rest, read_big_year_rest] =
```

Comment:
What do you think about this: every time we see a + or - sign, we use Integer.parse to extract the year and then we parse the rest (month/day) using the patterns below? Would that simplify things a bit? But the general direction is good to me.